tests/codegen/comparison-operators-2-tuple.rs

```diff
@@ -77,11 +80,17 @@ pub fn check_ge_direct(a: TwoTuple, b: TwoTuple) -> bool {
 // CHECK-SAME: (i16 noundef %[[A0:.+]], i16 noundef %[[A1:.+]], i16 noundef %[[B0:.+]], i16 noundef %[[B1:.+]])
 #[no_mangle]
 pub fn check_lt_via_cmp(a: TwoTuple, b: TwoTuple) -> bool {
-    // CHECK-DAG: %[[EQ:.+]] = icmp eq i16 %[[A0]], %[[B0]]
-    // CHECK-DAG: %[[CMP0:.+]] = icmp slt i16 %[[A0]], %[[B0]]
-    // CHECK-DAG: %[[CMP1:.+]] = icmp ult i16 %[[A1]], %[[B1]]
-    // CHECK: %[[R:.+]] = select i1 %[[EQ]], i1 %[[CMP1]], i1 %[[CMP0]]
-    // CHECK: ret i1 %[[R]]
+    // llvm-20-DAG: %[[CMP0:.+]] = tail call noundef range(i8 -1, 2) i8 @llvm.scmp.i8.i16(i16 %[[A0]], i16 %[[B0]])
+    // llvm-20-DAG: %[[EQ:.+]] = icmp eq i16 %[[A0]], %[[B0]]
+    // llvm-20-DAG: %[[CMP1:.+]] = tail call range(i8 -1, 2) i8 @llvm.ucmp.i8.i16(i16 %[[A1]], i16 %[[B1]])
+    // llvm-20: %[[S:.+]] = select i1 %[[EQ]], i8 %[[CMP1]], i8 %[[CMP0]]
+    // llvm-20: %[[R:.+]] = icmp eq i8 %[[S]], -1
```

Comment:
This is a regression. I'll take a look.

Reply:
https://github.com/llvm/llvm-project/pull/105536 should fix this.

Reply:
Nice! I'll wait for this to get submitted and update this PR.

Reply:
This is merged now.

Reply:
Nice! This eliminated the changes needed to one of the tests.